pkg/sql/parser/sql.y

```diff
@@ -5677,82 +5681,97 @@ show_backup_stmt:
       InCollection:    $4.expr(),
     }
   }
-| SHOW BACKUP string_or_placeholder opt_with_options
-  {
-    $$.val = &tree.ShowBackup{
-      Details: tree.BackupDefaultDetails,
-      Path:    $3.expr(),
-      Options: $4.kvOptions(),
-    }
-  }
+| SHOW BACKUP show_backup_kind FROM string_or_placeholder IN string_or_placeholder opt_with_options
+	{
+		$$.val = &tree.ShowBackup{
+			From:    true,
+			Kind:    *$3.showBackupKind(),
+			Path:    $5.expr(),
+			InCollection: $7.expr(),
+			Options: $8.kvOptions(),
+		}
+	}
 | SHOW BACKUP string_or_placeholder IN string_or_placeholder opt_with_options
-  {
-    $$.val = &tree.ShowBackup{
-      Details: tree.BackupDefaultDetails,
-      Path:    $3.expr(),
-      InCollection: $5.expr(),
-      Options: $6.kvOptions(),
-    }
-  }
+	{
+		$$.val = &tree.ShowBackup{
+			Kind:  tree.ShowBackupKind{
+								Details: tree.BackupDefaultDetails,
+							},
+			Path:    $3.expr(),
+			InCollection: $5.expr(),
+			Options: $6.kvOptions(),
+		}
+	}
+| SHOW BACKUP string_or_placeholder opt_with_options
+	{
+		$$.val = &tree.ShowBackup{
+		  Kind:    tree.ShowBackupKind{
+									Details: tree.BackupDefaultDetails,
+								},
+			Path:    $3.expr(),
+			Options: $4.kvOptions(),
+		}
+	}
 | SHOW BACKUP SCHEMAS string_or_placeholder opt_with_options
```

Comment:
I'm half tempted to suggest a breaking change if these were never doc'ed anyway, and require FROM between kind and string_or_placeholder here too, so that there are just two canonical forms:
a) SHOW BACKUP backup_loc_or_backup_in_collection opt_with_options
b) SHOW BACKUP backup_property FROM backup_loc_or_backup_in_collection opt_with_option


wdyt?

Reply:
maybe I don't understand your suggestion, but this line you're commenting on refers to old style SHOW BACKUP syntax, which we plan to deprecate. More broadly, this diff refactors sql.y to have 6 valid `SHOW BACKUP` incantations: 

 2 canonical forms for collections:
- `SHOW BACKUP show_backup_kind FROM string_or_placeholder IN string_or_placeholder opt_with_options`
    - the preferred, documented syntax
- `SHOW BACKUP string_or_placeholder IN string_or_placeholder opt_with_options`
    - only for valid default SHOW BACKUP.
    - no plans to deprecate, as that might break a lot workflows in production
    - not documented, as we want people to use FROM
    
4 for non collection SHOW BACKUP, which will have deprecation warnings in 22.1
- `SHOW BACKUP string_or_placeholder opt_with_options`
- `SHOW BACKUP SCHEMAS string_or_placeholder opt_with_options`
- `SHOW BACKUP RANGES string_or_placeholder opt_with_options`
- `SHOW BACKUP FILES string_or_placeholder opt_with_options`

Reply:
I was thinking maybe we leave just showing a backup, vs a property from/of that backup like schemas or files, as-is, without a FROM, and then `show_backup_kind` does not need an empty case, which generally makes things easier, and avoids having two syntaxes. It might also let us reuse the rule in the non-collection cases. 🤷

Reply:
ah, I see what you're saying. I'd like to keep my proposal as is because:
- I think it's worthwhile to include the FROM for the default show backup because RESTORE and SHOW syntax would then look quite similar. 
- SHOW BACKUP SCHEMAS is already documented, so if we go with your suggestion, we would need to document `SHOW BACKUP FROM SCHEMAS subdir IN dest` and `SHOW BACKUP subdir IN dest` which seems confusing.
- I don't think its worth adding new and breaking syntax changes for non collection SHOW BACKUP RANGES/SCHEMAS/FILES.  Who knows if users depend on these :/.

Reply:
👍